db/migrate/20210325004818_add_institute_reference_to_course_table.rb

```diff
@@ -0,0 +1,8 @@
+class AddInstituteReferenceToCourseTable < ActiveRecord::Migration
+  def change
+    # add foreign key institute_id column
+    change_table :courses do |t|
+      t.references :institute, null: false
```

Comment:
Does this work? I don't think we can enforce `null` here since we have existing courses. We'd need a default value if that were the case. 

Also worth noting that this only creates a `institute_id` column, and not an actual foreign key. You will need to use an external gem such as [foreigner](https://github.com/matthuhiggins/foreigner) to create FKs, but I think that is outside the scope of these tasks. FWIW, plus has it's own implementation of creating and managing foreign keys. 

For this, I think it's enough that we just create the reference and allow nulls at the database level, and in our code we can enforce presence with an activerecord validation (although - do we even need to enforce that?)

Reply:
This code seemed to work fine. But I didn't know about needing a gem to create a foreign key! I'd say we probably don't need to enforce presence via AR validation.

Reply:
Just for future reference, if we enforce `null: false` at the database level, we must also do the same at the active record level. Otherwise, active record will think a record is `valid?` since it only runs the "AR" validations, and then an error will be thrown by mysql when it attempts to save the record.